include/deal.II/base/utilities.h

```diff
@@ -483,8 +483,9 @@ namespace Utilities
    * calculations for constant expression arguments. The exponent @p iexp
    * must not be negative.
    */
-  constexpr unsigned int
-  pow(const unsigned int base, const int iexp)
+  template <typename T>
+  constexpr T
+  pow(const T base, const int iexp)
```

Comment:
Please add a comment that `T` must be an integral type (otherwise the initialization `1` might not work) and the case `pow <= 0 -> 1` is not true; it would be best if you add `static_assert(std::is_integral<T>::value, "Only integral types supported")` here.

Reply:
Thank! Done.